aptos-move/framework/aptos-framework/sources/transaction_validation.move

```diff
@@ -241,6 +278,36 @@ module aptos_framework::transaction_validation {
         );
     }
 
+    fun fee_payer_script_prologue_gas_deposit(
+        sender: signer,
+        txn_sequence_number: u64,
+        txn_sender_public_key: vector<u8>,
+        secondary_signer_addresses: vector<address>,
+        secondary_signer_public_key_hashes: vector<vector<u8>>,
+        fee_payer_address: address,
+        fee_payer_public_key_hash: vector<u8>,
+        txn_gas_price: u64,
+        txn_max_gas_units: u64,
+        txn_expiration_time: u64,
+        chain_id: u8,
+        required_deposit: Option<u64>,
+    ) {
+        fee_payer_script_prologue(
+            sender,
+            txn_sequence_number,
+            txn_sender_public_key,
+            secondary_signer_addresses,
+            secondary_signer_public_key_hashes,
+            fee_payer_address,
+            fee_payer_public_key_hash,
+            txn_gas_price,
+            txn_max_gas_units,
+            txn_expiration_time,
+            chain_id,
+        );
+        collect_deposit(fee_payer_address, required_deposit);
```

Comment:
nit: Shall we run this first?  I mean, if there's not enough balance we can save other computation.

Reply:
i'm not too sure. and actually current prologue checks balance so the order affects what error we return if the balance is insufficient.

I'm just thinking to diverge the flows between "with required deposit" and "without required deposit" as late as possible (and converge as early as possible)

https://github.com/aptos-labs/aptos-core/blob/7416ea343af11bda4df9eff3430a599d6622184c/aptos-move/framework/aptos-framework/sources/transaction_validation.move#L159

Reply:
ugh, okay. then document why you can't move it up (because the old epilogue checks account balance)

Reply:
Yeah we should run the prologue first for the checks it performs